src/openscm_runner/adapters/magicc7/_run_magicc_parallel.py

```diff
@@ -17,6 +19,43 @@
 LOGGER = logging.getLogger(__name__)
 
 
+class TemporaryDirectory:
```

Comment:
```suggestion
class TemporaryDirectoryIfNeeded:
```

Or something to differentiate from TemporaryDirectory ?

Reply:
Yes, got suggestion, I renamed it throughout.